synapse/lib/types.py

```diff
@@ -1903,6 +1903,11 @@ def _normPyList(self, valu):
             mesg = f'Tag does not match tagre: [{s_grammar.tagre.pattern}]'
             raise s_exc.BadTypeValu(valu=norm, name=self.name, mesg=mesg)
 
+        core = self.modl.core
+        if core is not None and not core.isTagValid(norm):
```

Comment:
FYI when adding a tag to a node, in snap._getTagNorm we end up calling core.isTagValid() twice.

Reply:
Yeah when attempting to add it we raise a `BadTag` exception and I wasn't sure if we wanted to change that behavior. `isTagValid` uses a cache so the second check should hit that anyways. `getTagNorm` in the snap is also cached so ideally we shouldn't be doing too much extra here.

Reply:
would it be safe to just snip the one in `_getTagNorm()` since it's dup?